src/os/i18n.zig

```diff
@@ -123,6 +126,26 @@ pub fn canonicalizeLocale(
     return buf[0..slice.len :0];
 }
 
+fn fixZhLocale(locale: []const u8) []const u8 {
```

Comment:
Add a doc comment explaining why (e.g. the libintl's internal canonicalize function returns erroneous results, etc)